### DISCUSSION TITLE
"Unbuildable" overlay?

### DISCUSSION THREAD

**Discussion by Ouaz** [Keriew/augustus#496] [Ideas]
As usual, I have no idea if it would require too much work, or if it's doable.

1) Starting from the Roads overlay (that hides everything), add an "Unbuildable" overlay that draws a grey cell on any unclearable/unbuildable tiles: rocks, water and coastlines (too small or not straight enough to build a dock, wharf, or shipyard).

2) Make the overlay relative to rocks, water, and coastlines (i.e. using the relative overlay shortcut)

**Why?**

Makes easier:
- to plan construction on narrow spaces or very rugged terrains
- to spot buildable tiles (especially at high DPI) whitout having to rotate the map and test tiles with the cursor.
- to spot straight and wide enough coastlines to build docks, wharves and shipyards
- to exploit buildable areas to the very last inches (OCD here )

Unbuildable cell png: ![cell_unbuildable](https://user-images.githubusercontent.com/11946570/135731170-fb7e6951-b0a7-4ad3-aaf3-8e1205ca62c3.png)

**- Miletus map** _(click to enlarge if needed)_

![miletus_normal_overlay](https://user-images.githubusercontent.com/11946570/135730905-686f9ec5-6c38-431d-9f1e-74e37ff7ba05.png)
![miletus_unbuildable_overlay](https://user-images.githubusercontent.com/11946570/136709686-50da836f-a02f-4cde-9642-3f44bfc39ffe.png)

Rocks and water are hidden. Trees and roads being clearable, they're not. Docks, wharfs, and shipyards being available in this mission, the coastlines where they can be built, are not hidden (3 and 2 tiles wide). In mission in which fishing is not available, only coastlines of 3 tiles wide would not be hidden (because suitable for docks only). 

**- Lugdunum map** _(click to enlarge if needed)_

![lugdunum_normal_overlay](https://user-images.githubusercontent.com/11946570/136709736-87df5b1a-06c5-4b79-90e9-501e4e7156e4.png)
![lugdunum_unbuildable_overlay](https://user-images.githubusercontent.com/11946570/136709742-ae6a7196-4a65-4f7f-aa02-8b21e8dc95fe.png)

The unbuildable overlay makes easier to manage construction near the cliffs (preventing some mistakes or unexpected unclearable tiles), and on the high and narrow fertile land.

----------------------------------

As usual, let me know what you think about it.

Thanks for reading.

**Comment by crudelios** [comment]
This could work, except for the docks thing (showing all lakes as unbuildable). It's technically possible to implement but it's a heavy operation, and also technically, you can build a dock/wharf there, only it will be inoperational.
